### PR TITLE
Visualization: add averaged stress/strain option

### PR DESCRIPTION
**Pull request by tjhei**
Introduce the setting "Point-wise stress and strain" to average
variables related to stress and strain in visualization.

Note: as discussed in the biweekly ASPECT meeting, this changes the default to average the values.

alternative to #3472

### CODE REVIEW

**Comment by tjhei** [comment]
I added a second version for each test that depended on point-wise output to have both versions around.

File: include/aspect/postprocess/visualization.h [hunk 1]
```diff
@@ -368,6 +387,13 @@ namespace aspect
         void
         write_plugin_graph (std::ostream &output_stream);
 
+
```
Comment (gassmoeller) [inline include/aspect/postprocess/visualization.h:390]: remove additional empty line

File: include/aspect/postprocess/visualization.h [hunk 2]
```diff
@@ -454,6 +480,14 @@ namespace aspect
          */
         bool filter_output;
 
+
```
Comment (gassmoeller) [inline include/aspect/postprocess/visualization.h:483]: same here

File: include/aspect/postprocess/visualization/averaged_strain_rate.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (gassmoeller) [inline include/aspect/postprocess/visualization/averaged_strain_rate.h:1]: Is this file still needed, I thought we would use the existing postprocessor?

File: source/postprocess/visualization/averaged_strain_rate.cc [hunk 4]
```diff
@@ -0,0 +1,118 @@
+/*
```
Comment (gassmoeller) [inline source/postprocess/visualization/averaged_strain_rate.cc:1]: same here, is this file needed?

**Comment by tjhei** [comment]
thanks for the review. Fixed!
